src/die_init.cpp

```diff
@@ -332,17 +341,33 @@ namespace Die
                 // Start advertising!
                 Stack::startAdvertising();
 
+                // Initialize common logic
+                initMainLogic();
+
             #if defined(DEBUG_FIRMWARE)
-                initDebugLogic();
+                if (!inValidation) 
+                {
+                    // Initialize debug logic
+                    initDebugLogic();
+                }
                 NRF_LOG_INFO("---------------");
             #else
-                // Initialize main logic manager
-                initMainLogic();
+                // Initialize main die logic
+                if (!inValidation) 
+                {
+                    initDieLogic();
+                }
                 NRF_LOG_INFO("---------------");
 
                 // Entering the main loop! Play Hello! anim
-                // if not using loop anim and not resetting from sleep
-                if (!loopAnim) {
+                // if in validation mode
+                if (inValidation) {
+                    ValidManager::init();
+                    ValidManager::onDiceInitialized();
+                }
+                // if not using loop anim
+                else if (!loopAnim) {
+                    AnimController::hook(feed, nullptr);
```

Comment:
The call to `AnimController::hook(feed, nullptr);` is made twice (but in different conditions), please simplify to have only one call